Azure EventHub Consume and Trigger task: checkpointStoreProperties marked as not required, but the task fails if this attribute is not present

### Expected Behavior

For Azure EventHub Consume and Trigger task, checkpointStoreProperties should either be a required field, OR if it marked not required, then the task should succeed when this property is absent.

### Actual Behaviour

When checkpointStoreProperties is not provided, we get the following error:

```
2024-03-25 15:23:51.395 Invalid URL format. URL: null
2024-03-25 15:23:51.395 Cannot invoke "String.length()" because "<parameter2>" is null
Cannot invoke "String.length()" because "<parameter2>" is null
2024-03-25 15:23:51.395 java.lang.IllegalArgumentException: Invalid URL format. URL: null
	at com.azure.storage.blob.BlobUrlParts.parse(BlobUrlParts.java:349)
	at com.azure.storage.blob.implementation.util.BuilderHelper.httpsValidation(BuilderHelper.java:188)
	at com.azure.storage.blob.implementation.util.BuilderHelper.buildPipeline(BuilderHelper.java:119)
	at com.azure.storage.blob.BlobContainerClientBuilder.buildAsyncClient(BlobContainerClientBuilder.java:163)
	at io.kestra.plugin.azure.eventhubs.client.EventHubClientFactory.createBlobContainerAsyncClient(EventHubClientFactory.java:146)
	at io.kestra.plugin.azure.eventhubs.Consume.getBlobCheckpointStore(Consume.java:217)
	at io.kestra.plugin.azure.eventhubs.Consume.run(Consume.java:141)
	at io.kestra.plugin.azure.eventhubs.Consume.run(Consume.java:117)
	at io.kestra.plugin.azure.eventhubs.Consume.run(Consume.java:48)
	at io.kestra.core.runners.Worker$WorkerThread.run(Worker.java:710)
Caused by: java.net.MalformedURLException: Cannot invoke "String.length()" because "<parameter2>" is null
	at java.base/java.net.URL.<init>(Unknown Source)
	at java.base/java.net.URL.<init>(Unknown Source)
	at java.base/java.net.URL.<init>(Unknown Source)
	at com.azure.storage.blob.BlobUrlParts.parse(BlobUrlParts.java:347)
	... 9 more
Caused by: java.lang.NullPointerException: Cannot invoke "String.length()" because "<parameter2>" is null
	... 13 more
```

### Steps To Reproduce

1. Run the flow provided in the Example Flow. (Ensure you do not add `checkpointStoreProperties` property)
2. Check the logs on executing this flow.

### Environment Information

- Kestra Version: 0.15.5
- Plugin version: 0.15.5
- Operating System (OS / Docker / Kubernetes): Docker
- Java Version (If not docker):


### Example flow

```
id: ConsumeDataEventsFromAzureEventHubs
namespace: company.team
tasks:
- id: consumeFromEventHubs
  type: io.kestra.plugin.azure.eventhubs.Consume
  eventHubName: <event-name>
  namespace: <eventhub-namespace>
  connectionString: "<connectionString>"
  bodyDeserializer: STRING
  consumerGroup: "$Default"
```

--- a/src/main/java/io/kestra/plugin/azure/eventhubs/Consume.java
+++ b/src/main/java/io/kestra/plugin/azure/eventhubs/Consume.java
@@ -24,6 +24,7 @@ import io.kestra.plugin.azure.eventhubs.service.consumer.EventHubConsumerService
 import io.kestra.plugin.azure.eventhubs.service.consumer.EventHubNamePartition;
 import io.kestra.plugin.azure.eventhubs.service.consumer.StartingPosition;
 import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
 import lombok.AccessLevel;
 import lombok.Builder;
 import lombok.EqualsAndHashCode;
@@ -102,8 +103,8 @@ public class Consume extends AbstractEventHubTask implements EventHubConsumerInt
     @Builder.Default
     private Property<Duration> maxDuration = Property.ofValue(Duration.ofSeconds(10));
 
-    @Builder.Default
-    private Property<Map<String, String>> checkpointStoreProperties = Property.ofValue(new HashMap<>());
+    @NotNull
+    private Property<Map<String, String>> checkpointStoreProperties;
 
     // SERVICES
     @Getter(AccessLevel.NONE)
@@ -215,6 +216,22 @@ public class Consume extends AbstractEventHubTask implements EventHubConsumerInt
                                                    final EventHubConsumerInterface pluginConfig,
                                                    final EventHubClientFactory factory) throws IllegalVariableEvaluationException {
         var renderedMap = runContext.render(pluginConfig.getCheckpointStoreProperties()).asMap(String.class, String.class);
+
+        var connectionString = renderedMap.get("connectionString");
+        var containerName = renderedMap.get("containerName");
+
+        if (connectionString == null || connectionString.trim().isEmpty()) {
+            throw new IllegalArgumentException(
+                "checkpointStoreProperties.connectionString is required. Provide your Azure Blob Storage connection string."
+            );
+        }
+
+        if (containerName == null || containerName.trim().isEmpty()) {
+            throw new IllegalArgumentException(
+                "checkpointStoreProperties.containerName is required. Provide your Azure Blob Storage container name."
+            );
+        }
+
         BlobContainerClientInterface config = BlobContainerClientInterface.builder()
             .containerName(Property.ofValue(renderedMap.get("containerName")))
             .connectionString(Property.ofValue(renderedMap.get("connectionString")))

--- a/src/main/java/io/kestra/plugin/azure/eventhubs/Trigger.java
+++ b/src/main/java/io/kestra/plugin/azure/eventhubs/Trigger.java
@@ -10,6 +10,7 @@ import io.kestra.core.runners.RunContext;
 import io.kestra.plugin.azure.eventhubs.serdes.Serdes;
 import io.kestra.plugin.azure.eventhubs.service.consumer.StartingPosition;
 import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
 import lombok.*;
 import lombok.experimental.SuperBuilder;
 import lombok.extern.slf4j.Slf4j;
@@ -105,8 +106,8 @@ public class Trigger extends AbstractTrigger implements EventHubConsumerInterfac
     @Builder.Default
     private Property<Duration> maxDuration = Property.ofValue(Duration.ofSeconds(10));
 
-    @Builder.Default
-    private Property<Map<String, String>> checkpointStoreProperties = Property.ofValue(new HashMap<>());
+    @NotNull
+    private Property<Map<String, String>> checkpointStoreProperties;
 
     private Property<String> namespace;
 

